p2p/agent.go

```diff
@@ -125,6 +134,14 @@ type (
 		reconnectTimeout           time.Duration
 		reconnectTask              *routine.RecurringTask
 		qosMetrics                 *Qos
+		networkSeparator           *networkSeparator
+	}
+
+	messageNetworkMappingFunc func(proto.Message) (string, error)
+
+	networkSeparator struct {
+		networks     []string
```

Comment:
topicSeparator struct {
    topics []string
}

Reply:
i use another way, pls review the next commit